chore/issue 160 renaming test experimental scripts to jsx

hrushi5387 opened the pull request:
<!--
## Submitting a Pull Request
We love contributions and appreciate any help you can offer!
-->

## Related Issue
<!-- Include a link to the issue (e.g. Resolves #12) -->
Resolves https://github.com/ProjectEvergreen/wcc/issues/160

## Summary of Changes
<!-- Briefly summarize the changes made, lists are appreciated, ideally with checklists-->

1. [x] Rename test-exp-loader.js -> test-loader.js (and update naming in npm scripts). addition to this I have also renamed test-exp-register.js -> test-register.js as this file which is relevant to experimental testing.
2. [x] Rename all applicable npm scripts from test:exp -> test:jsx
3. [x] Update any script names in .github/workflows - there was no mention of script names in workflow files, but I have updated test:jsx change in these files

thescientist13 commented on .github/workflows/ci-exp-win.yml:
```diff
@@ -25,7 +25,7 @@ jobs:
     - name: Test
       run: |
-        npm run test:exp
+        npm run test:jsx
```
For consistency we would want to rename these GitHub Action files as well
- _ci-exp-win.yml_ -> _ci-jsx-win.yml_
- _ci-exp.yml_ -> _ci-jsx.yml_

hrushi5387 replied:
this is also done

thescientist13 reviewed (commented):
Thanks for this!

Left a couple small follow-ups for you, but otherwise this is looking good so far.  👍

hrushi5387 commented:
Hi @thescientist13 ,
I have done following changes as per above comment. Could you please check and confirm?

- [x] changed test:tdd:exp -> test:tdd:jsx in package.json file
- [x] renamed ci-exp-win.yml -> ci-jsx-win.yml  in .github/workflow folder
- [x] renamed ci-exp.yml -> ci-jsx.yml in .github/workflow folder

Regards,
Hrushi

hrushi5387 commented on package.json:
```diff
@@ -38,12 +38,12 @@
+    "test:jsx": "c8 node --import ./test-register.js --experimental-strip-types ./node_modules/mocha/bin/mocha \"./test/**/**/*.spec.js\"",
     "test:tdd": "npm run test -- --watch",
-    "test:tdd:exp": "npm run test:exp -- --watch"
+    "test:tdd:exp": "npm run test:jsx -- --watch"
```
done

thescientist13 reviewed (approved):
Thanks!
